pkg/hintrunner/zero/zerohint_dictionaries.go

```diff
@@ -191,6 +194,80 @@ func createDictReadHinter(resolver hintReferenceResolver) (hinter.Hinter, error)
 	return newDictReadHint(dictPtr, key, value), nil
 }
 
+// DictSquashCopyDict hint prepares arguments for creating a new dictionary
+// with `dict_new` by copying the existing one in the context of squashing
+//
+// `newDictSquashCopyDictHint` takes 1 operander as argument
+//   - `dict_accesses_end` variable is a memory address that allows access to the dictionary
+//
+// `newDictSquashCopyDictHint`assigns `initial_dict` to the retrieved dictionary in the current scope
+func newDictSquashCopyDictHint(dictAccessesEnd hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "DictSquashCopyDict",
+		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+			//> # Prepare arguments for dict_new. In particular, the same dictionary values should be copied
+			//> # to the new (squashed) dictionary.
+			//> vm_enter_scope({
+			//> 	# Make __dict_manager accessible.
+			//> 	'__dict_manager': __dict_manager,
+			//> 	# Create a copy of the dict, in case it changes in the future.
+			//> 	'initial_dict': dict(__dict_manager.get_dict(ids.dict_accesses_end)),
+			//> })
+
+			dictionaryManager_, err := ctx.ScopeManager.GetVariableValue("__dict_manager")
+			if err != nil {
+				return err
+			}
+
+			dictionaryManager, ok := dictionaryManager_.(hinter.ZeroDictionaryManager)
+			if !ok {
+				return fmt.Errorf("cannot cast dictionaryManager_ to a DictionaryManager struct")
+			}
```

Comment:
I think there is a utility for this, `GetZeroDictionaryManager` or something like that

Reply:
what do you mean? i'm just type casting here

Reply:
@cicr99  just addressed your comment, thanks :)